interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java

```diff
@@ -295,14 +295,14 @@ private XdsStatsWatcher(long startId, long endId) {
       this.endId = endId;
     }
 
-    void rpcCompleted(long requestId, @Nullable String serverId) {
+    void rpcCompleted(long requestId, @Nullable String hostname) {
       synchronized (lock) {
         if (startId <= requestId && requestId < endId) {
-          if (serverId != null) {
-            if (rpcsByPeer.containsKey(serverId)) {
-              rpcsByPeer.put(serverId, rpcsByPeer.get(serverId) + 1);
+          if (hostname != null) {
+            if (rpcsByPeer.containsKey(hostname)) {
+              rpcsByPeer.put(hostname, rpcsByPeer.get(hostname) + 1);
```

Comment:
You might need use a generated random hostname as a fallback hostname in `XdsTestServer`.  As in https://github.com/grpc/grpc-java/commit/e7d7c5bde4caf606f02b2575d5f62e80ad5135b0#diff-2140bcb433e6510183658de7492bfb57R58

Reply:
I think it's WAI as-is: the test driver script will need to assert on the hostnames recorded by the client, and so a randomly generated host won't be useful - if hostname is missing, the test will (and should) be failed.

Reply:
If the hostname is missing, how will test fail? Is it easy to debug then?

Reply:
The test spec is still in progress, but a sample scenario is something like: there are four backends, and we expect the client to round-robin over all four. If one of the hostnames is missing, the test will see that it is round-robining over three backends and fail.

Reply:
Good point about debugging. Modified server to throw (fail) if it can't get hostname.